### PR TITLE
CF tested and working for EC2 deployment

### LINKED ISSUE #132: EC2 AWS: Cloudformation
In a project to implement Algo as a native mobile application to both iOS and Android it was necessary to migrate much of the AWS infrastructure deployment for Algo into a [Cloudformation](https://aws.amazon.com/cloudformation/) template to eliminate dependencies, simplify development and reduce the required infrastructure to deploy Algo as a mobile application.

There are a few benefits over using Cloudformation on AWS rather than programmatically managing the resources necessary for Algo as we are currently doing.

- All resources are created, owned and managed by the stack owner.
- All resources are unique named to prevent naming collisions and promote a multi-user environment.
- Better resource limit management and cost estimation.
- Destroying a stack removes ALL resources associated with that instance of Algo.

In short, it simplifies managing all the necessary resources for the end user and prevents mistakes. It may be worth looking into the plausibility of reusing the template for Algo itself in the future.

### PR DESCRIPTION
**Pull request by aboutte**
I took the [CloudFormation](https://github.com/trailofbits/algo/tree/CloudFormation) branch and fixed up a couple of things to get CF working for the EC2 deployment.

This PR can close  #132.

### CODE REVIEW

**Comment by gunph1ld** [comment]
Agree, I think we need to handle IPv6 before merge, I made some drafts in the branch, but not sure if it works. I will check tonight

**Comment by gunph1ld** [comment]
I'm not sure, but seems the [AWS docs](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-vpc.html) are wrong, I can not get the Ipv6CidrBlocks value form `AWS::EC2::VPC`

```
Outputs:
  Ipv6Cidr:
    Value: !GetAtt VPC.Ipv6CidrBlocks
```

The errors is:
`Template format error: The Value field of every Outputs member must evaluate to a String.`

**Comment by gunph1ld** [comment]
New things implemented. @aboutte could you rebase with [the initial branch](https://github.com/trailofbits/algo/tree/CloudFormation)?

**Comment by gunph1ld** [comment]
Nice. Fixes #60 also

**Comment by aboutte** [comment]
@gunph1ld this is tested and working now.  I had to make a few changes to get ipv6 working correctly.  Because we have ipv6 forwarding turned on I had to add `sysctl net.ipv6.conf.\$IFACE.accept_ra=2`, otherwise I was not getting an ipv6 default route.  

screen shot 2017-04-18 at 10 17 25 am

**Comment by dguido** [comment]
Awesome!

File: roles/cloud-ec2/templates/stack.yml.j2 [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+      - InstanceSecurityGroup
+    Metadata:
+      AWS::CloudFormation::Init:
+        config:
+          users:
+            ubuntu:
+              groups:
+                - "sudo"
+              homeDir: "/home/ubuntu/"
+          files:
```
Comment (defunctio) [inline roles/cloud-ec2/templates/stack.yml.j2]: There is nothing wrong with this method, but I'm just wondering if there was a specific reason for this vs using ssh-authorized-keys param?

**Pull request merged**